django/thunderstore/repository/views/repository.py

```diff
@@ -444,6 +444,8 @@ def get_object(self, *args, **kwargs):
         )
         if not listing.can_be_viewed_by_user(self.request.user):
             raise Http404("Package is waiting for approval or has been rejected")
+        if not listing.package.is_active:
+            raise Http404("Package has been disabled")
```

Comment:
Should return a standard 404 instead of a special message, as this leaks information (package confirmed to have existed) which might be unwanted.

Reply:
I've added this back since, this isn't shown to the user. It's catched by the middleware and a 404 page is displayed to the user.